api/layer/object.go

```diff
@@ -363,22 +356,21 @@ func (n *layer) listSortedAllObjects(ctx context.Context, p allObjectParams) ([]
 	return objects, nil
 }
 
-func trimStartAfter(startAfter string, objects []*ObjectInfo) []*ObjectInfo {
-	if objects != nil && len(startAfter) != 0 && objects[0].Name <= startAfter {
-		for i := range objects {
-			if objects[i].Name > startAfter {
-				return objects[i:]
-			}
-		}
-	}
-	return objects
+func (n *layer) trimAndCacheRestObjects(cacheKey string, maxKeys int, objects []*ObjectInfo) []*ObjectInfo {
+	restObjects := objects[maxKeys-1:]
+	n.cache.Put(cacheKey, restObjects)
+
+	return objects[:maxKeys]
 }
 
-func trimAfterObjectID(id string, objects []*ObjectInfo) []*ObjectInfo {
-	for i, obj := range objects {
-		if obj.ID().String() == id {
-			return objects[i:]
+func sortObjects(allObjects []*ObjectInfo) (prefixes []string, objects []*ObjectInfo) {
```

Comment:
A bit confusing name. Maybe is it better to use the word "filter" or "sort"?

Reply:
I meaned sorting objects into separate lists (objects and prefixes), but triage sounds better, renamed.